kotlinx-coroutines-core/jvm/src/debug/internal/DebugProbesImpl.kt

```diff
@@ -163,6 +163,50 @@ internal object DebugProbesImpl {
                 }
         }
 
+    /*
+     * This method optimises the number of packages sent by the IDEA debugger
+     * to a client VM to speed up fetching of coroutine information.
+     * The return value is an array of objects, which consists of four elements:
+     * 1) A string in a JSON format that stores information that is needed to display
+     *    every coroutine in the coroutine panel in the IDEA debugger.
+     * 2) An array of last observed threads.
+     * 3) An array of last observed frames.
+     * 4) An array of DebugCoroutineInfo.
+     */
+    @OptIn(ExperimentalStdlibApi::class)
+    public fun dumpCoroutinesInfoAsJSONAndReferences(): Array<Any> {
```

Comment:
```suggestion
    public fun dumpCoroutinesInfoAsJsonAndReferences(): Array<Any> {
```

Reply:
Please add a few unit-tests that ensure the resulting format, for empty list of coroutines and for some coroutines (mixing `null` and non-null values)